raw_result_diff.py

```diff
@@ -0,0 +1,198 @@
+                summary[k] = 0
+
+        for result in self.new_tests.values():
+            self.summary["new_tests"][result] += 1
+
+        for result in self.removed_tests.values():
+            self.summary["removed_tests"][result] += 1
+
+        for result in self.diff_tests.values():
+            self.summary["diff_tests"][result["old_result"]] -= 1
+            self.summary["diff_tests"][result["new_result"]] += 1
+
+        # Calculate summary column widths to make the summary look better
+        for v in TestResult:
+            self.summary_column_widths[v] = max(
+                len(str(self.summary["new_tests"][v])) + 1,
+                len(str(self.summary["removed_tests"][v])) + 1,
+                len(str(self.summary["diff_tests"][v])) + 1,
+            )
+
+    def print_summary_results(self, summary_map: dict) -> None:
+        for v in TestResult:
+            print(
+                f"{summary_map[v]:+{self.summary_column_widths[v]}d} {EMOJIS[v]}   ",
+                end="",
+            )
+
+    def print_full_results(self) -> None:
+        print("Duration:")
+        print(f"     {self.duration_delta:+.2f}s")
+
+        has_new_tests = len(self.new_tests) > 0
+        has_removed_tests = len(self.removed_tests) > 0
+        has_diff_tests = len(self.diff_tests) > 0
+
+        if not has_new_tests and not has_removed_tests and not has_diff_tests:
+            return
+
+        print()
+        print("Summary:")
+
+        if has_new_tests:
+            print("    New Tests:\n        ", end="")
+            self.print_summary_results(self.summary["new_tests"])
+            print()
+
+        if has_removed_tests:
+            print("    Removed Tests:\n        ", end="")
+            self.print_summary_results(self.summary["removed_tests"])
+            print()
+
+        if has_diff_tests:
+            print("    Diff Tests:\n        ", end="")
+            self.print_summary_results(self.summary["diff_tests"])
+            print()
+
+        print()
+
+        if has_new_tests:
+            print("New Tests:")
+            for path, result in self.new_tests.items():
+                print(f"    {path:{self.longest_path_length}s} {EMOJIS[result]}")
+            print()
+
+        if has_removed_tests:
+            print("Removed Tests:")
+            for path, result in self.removed_tests.items():
+                print(f"    {path:{self.longest_path_length}s} {EMOJIS[result]}")
+            print()
+
+        if has_diff_tests:
+            print("Diff Tests:")
+            for path, result in self.diff_tests.items():
+                old_emoji = EMOJIS[result["old_result"]]
+                new_emoji = EMOJIS[result["new_result"]]
+                print(
+                    f"    {path:{self.longest_path_length}s} {old_emoji} -> {new_emoji}"
+                )
+
+    def print_regressions(self) -> None:
+        for (
+            path,
+            result,
+        ) in self.diff_tests.items():
```

Comment:
```suggestion
        for path, result in self.diff_tests.items():
```

Reply:
Running `black` did this for some reason

Reply:
thats becasue you wrapped them in parentheses, which is unnecessary

Reply:
It will always force multiple line if a trailing comma is present (`result,`), but shouldn't do that otherwise - i.e. it should accept the proposed change without reformatting it.